load-tsfile-tool/src/main/java/org/apache/iotdb/RewriteFileTool.java

```diff
@@ -0,0 +1,366 @@
+ * insertTablet() for every page in chunk, in case chunk is too large. `Move` command is used to
+ * unload files in iotdb, and mods files are moved manually.
+ */
+public class RewriteFileTool {
+  // backup data dir path
+  private static String backUpDirPath;
+  // validation file path
+  private static String validationFilePath;
+  // tsfile list path
+  private static String tsfileListPath;
+  // output file path
+  private static String outputLogFilePath;
+
+  private static final String HostIP = "localhost";
+  private static final String rpcPort = "6667";
+  private static String user = "root";
+  private static String password = "root";
+
+  private static final FSFactory fsFactory = FSFactoryProducer.getFSFactory();
+  private static final long MAX_TABLET_SIZE = 1024 * 1024;
+
+  private static PrintWriter writer;
+
+  /**
+   * -b=[path of backUp directory] -vf=[path of validation file]/-f=[path of tsfile list] -o=[path
+   * of output log] -u=[username, default="root"] -pw=[password, default="root"]
+   */
+  public static void main(String[] args) throws IOException {
+    if (!checkArgs(args)) {
+      System.exit(1);
+    }
+    writer = new PrintWriter(new FileWriter(outputLogFilePath));
+    try {
+      if (validationFilePath != null) {
+        readValidationFile(validationFilePath);
+      }
+      if (tsfileListPath != null) {
+        readTsFileList(tsfileListPath);
+      }
+    } catch (Exception e) {
+      printBoth(e.getMessage());
+      e.printStackTrace();
+    } finally {
+      writer.close();
+    }
+  }
+
+  public static void readValidationFile(String validationFilePath)
+      throws IOException, IoTDBConnectionException {
+    Session session = new Session(HostIP, rpcPort, user, password);
+    session.open(false);
+
+    BufferedReader bufferedReader = new BufferedReader(new FileReader(validationFilePath));
+    String line;
+    while ((line = bufferedReader.readLine()) != null) {
+      if (!line.startsWith("-- Find the bad file ")) {
+        continue;
+      }
+      String badFilePath = line.replace("-- Find the bad file ", "");
+      unloadAndReWriteWrongTsFile(badFilePath, session);
+    }
+    bufferedReader.close();
+    session.close();
+    printBoth("Finish rewriting all bad files.");
+  }
+
+  public static void readTsFileList(String tsfileListPath)
+      throws IoTDBConnectionException, IOException {
+    Session session = new Session(HostIP, rpcPort, user, password);
+    session.open(false);
+
+    BufferedReader bufferedReader = new BufferedReader(new FileReader(tsfileListPath));
+    String badFilePath;
+    while ((badFilePath = bufferedReader.readLine()) != null) {
+      unloadAndReWriteWrongTsFile(badFilePath, session);
+    }
+    bufferedReader.close();
+    session.close();
+    printBoth("Finish rewriting all bad files.");
+  }
+
+  public static void unloadAndReWriteWrongTsFile(String filename, Session session) {
```

Comment:
The TsFile to be rewrited is not managed by IoTDB, no need to unload.

Reply:
I delete all rewriteFile related.